packages/plugins/posthog-ui/src/lib/interceptors/posthog.interceptor.ts

```
@@ -0,0 +1,140 @@
+ * HTTP interceptor that automatically tracks API requests using PostHog.
+ * Captures duration, status, method, and error info (if any).
+ */
+@Injectable()
+export class PostHogInterceptor implements HttpInterceptor {
+	private config: PostHogModuleConfig;
+	private debugMode: boolean;
+
+	constructor(
+		private posthogServiceManager: PostHogServiceManager,
+		@Inject(POSTHOG_CONFIG) config: PostHogModuleConfig,
+		@Optional() @Inject(POSTHOG_DEBUG_MODE) debugMode: boolean = false
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)